### PR TITLE
Use java 1.8 or greater for fastrtpsgen.jar generation [5666]

### LINKED ISSUE #550: fastrtpsgen doesn't compile due to end of Java 6 support [5665]
Attempting to build Fast-RTPS v1.7.2 on Windows 10 in a CMD shell results in a build error during the `java` build target:

```
%1 not a valid Win32 application
```
The error seems related to a gradle compile failure

Commands used:

```
> set INSTDIR=%cd%/install
> cd Fast-RTPS-1.7.2
> mkdir build
> cd build
> cmake -DTHIRDPARTY=ON -DBUILD_JAVA=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=%INSTDIR% ..
> cmake --build . --target install
```

Environment used:
```
> echo %PATH%
C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\bin\Roslyn;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools\;C:\Program Files (x86)\Windows Kits\10\bin\10.0.17763.0\x86;C:\Program Files (x86)\Windows Kits\10\bin\x86;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\\MSBuild\15.0\bin;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\;C:\Qt\5.12.3\msvc2017\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\dotnet\;C:\Program Files\Java\java-se-8u40-ri\bin;C:\Program Files\CMake\bin;C:\Program Files\Git\cmd;C:\Program Files\7-Zip;C:\Gradle\gradle-5.4.1\bin;C:\MinGW\msys\1.0\bin;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja

> echo %JAVA_HOME%
C:\Program Files\Java\java-se-8u40-ri

> gradle --version

------------------------------------------------------------
Gradle 5.4.1
------------------------------------------------------------

Build time:   2019-04-26 08:14:42 UTC
Revision:     261d171646b36a6a28d5a19a69676cd098a4c19d

Kotlin:       1.3.21
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          1.8.0_40 (Oracle Corporation 25.40-b25)
OS:           Windows 8.1 6.3 x86

> java -version
openjdk version "1.8.0_40"
OpenJDK Runtime Environment (build 1.8.0_40-b25)
OpenJDK Client VM (build 25.40-b25, mixed mode)
```

Fast-RTPS version used include 

- v1.8.0 tag 
- v1.7.2 tag

Java SE versions used include 
- Oracle 8u212 
- Oracle 12.0.1
- openJDK 1.8.0_40 (windows)

CMake generators used include 

- Visual Studio 15 2017 Win64
- Nmake makefiles
- Ninja
- Default (no -G)

### PR DESCRIPTION
**Pull request by breezZe**
If we use a java version higher than java 6, problems will be like #555 and #550 . 
The method for TemplateManager tmanager has been changed, so it needed be updated.
Now if you use openjdk-1.8 and gradle 5.4.1, everything is fine.

*Remember to fix file /thirdparty/idl/idl.gradle*
`compileJava { sourceCompatibility = 1.8 targetCompatibility = 1.8 }`

Fixes #550

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1868/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1827/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1839/)

File: fastrtpsgen/src/main/java/com/eprosima/fastrtps/fastrtpsgen.java [hunk 1]
```diff
@@ -522,7 +522,7 @@ private Project parseIDL(String idlFilename) {
             topicann.addMember(new AnnotationMember("value", new PrimitiveTypeCode(Kind.KIND_BOOLEAN), "true"));
 
             // Create template manager
-            TemplateManager tmanager = new TemplateManager("FastCdrCommon:eprosima:Common");
+            TemplateManager tmanager = new TemplateManager("FastCdrCommon:eprosima:Common",ctx,true);
```
Comment (LuisGP) [inline fastrtpsgen/src/main/java/com/eprosima/fastrtps/fastrtpsgen.java]: This new usage of TemplateManager will receive **false** by default.

```suggestion
            TemplateManager tmanager = new TemplateManager("FastCdrCommon:eprosima:Common", ctx, false);
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1908/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1867/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1880/)

**Comment by richiware** [comment]
Failed tests  not related.

**Pull request merged**